src/python/pants/option/arg_splitter.py

```diff
@@ -257,13 +257,15 @@ def _descope_flag(self, flag, default_scope):
           return scope, flag_prefix + flag[len(prefix):]
     return default_scope, flag
 
-  def _at_flag(self):
-    return (self._unconsumed_args and
-            self._unconsumed_args[-1].startswith('-') and
-            not self._at_double_dash())
+  def _at_flag(self) -> bool:
+    return (
+      bool(self._unconsumed_args)
```

Comment:
This was an actual bug, or at least it wasn't working as intended. Without the wrapping `bool`, this function was sometimes returning `[]` and only sometimes `False`. Now, it always returns a bool.